service/controller/resource/app/desired.go

```diff
@@ -302,6 +329,29 @@ func (r *Resource) newAppSpecs(ctx context.Context, cr apiv1alpha2.Cluster) ([]k
 	return specs, nil
 }
 
+func newAppOperatorAppSpec(cr apiv1alpha2.Cluster, component releaseversion.ReleaseComponent) key.AppSpec {
+	var operatorAppVersion string
+
+	// Setting the reference allows us to deploy from a test catalog.
+	if component.Reference != "" {
+		operatorAppVersion = component.Reference
```

Comment:
Setting the reference makes it easy to test app-operator via the release CR. 🎉 e.g.  

```yaml
 components:
  - catalog: control-plane-test-catalog
    name: app-operator
    reference: 4.0.0-80a5b1e2cd87675a7127d18cce8a14de8463a6c7
    version: 4.0.1-dev
```